tests/10-EdvClient.spec.js

```diff
@@ -16,6 +16,106 @@ describe('EdvClient', () => {
     await mock.server.shutdown();
   });
 
+  it('should only contain one indexed document after 3 updates', async () => {
+    const getLatestDocsByIndexHas = async (client, invocationSigner, index) => {
+      const indexedDocs = await client.find({
+        has: index,
+        invocationSigner
+      });
+      const uniqueIds = [...new Set(indexedDocs.map(item => item.id))];
+
+      const docsFromGet = await Promise.all(
+        uniqueIds.map(id => {
+          return client.get({id, invocationSigner});
+        })
+      );
+      return docsFromGet;
+    };
+
+    const client = await mock.createEdv();
+    client.ensureIndex({attribute: 'content.indexedKey'});
+    const testId = await EdvClient.generateId();
+    const doc = {
+      id: testId,
+      content: {someKey: '111111', indexedKey: 'value1'}
+    };
+    let version1 = await client.insert({
+      doc,
+      invocationSigner,
+      keyResolver
+    });
+    version1 = await client.get({id: doc.id, invocationSigner});
+
+    // console.log("version: 1", version1);
+    await client.update({
+      doc: {
+        ...version1,
+        content: {
+          ...version1.content,
+          someKey: '22222'
+        }
+      },
+      invocationSigner,
+      keyResolver
+    });
+    const version2 = await client.get({id: doc.id, invocationSigner});
+    // console.log("version: 2", version2);
+    await client.update({
+      doc: {
+        ...version2,
+        content: {
+          ...version2.content,
+          someKey: '33333'
+        }
+      },
+      invocationSigner,
+      keyResolver
+    });
+
+    const docsFromGet = await getLatestDocsByIndexHas(
```

Comment:
My original confusion was whether the result of this custom function should be equal to the direct client call:

```
const indexedDocs = await client.find({
        has: 'content.indexedKey,
        invocationSigner
      });
```

Reply:
If its the case that the index should return 1 document, you should remove my hacky getLatestDocsByIndexHas function from this test.

Reply:
@OR13 this removes your hack function https://github.com/digitalbazaar/edv-client/pull/18/commits/91e6ac4cf2d27970276c9e42eec616268dd7c13f

However one of your tests should be failing.
When you update an EDV document you either update the whole thing or nothing.
So your smaller test where on update you do not have `indexedKey` in the content on update means the there is no longer an `indexedKey` for that document. So it should have been failing, but was failing for the wrong reason! Basically EDV documents do not support partial updates the way mongo documents normally do. Primary reason for this is probably because your document is stored encrypted in the database and mongo does not have operators that can work on encrypted data yet. Also it's just simple and easy to manage.

Reply:
Thanks, this make things much clearer, when performing an update, any indexes that need to be applied need to be maintained, manually.